server/src/main/java/io/whitefox/core/services/S3FileSigner.java

```diff
@@ -0,0 +1,58 @@
+package io.whitefox.core.services;
+
+import io.whitefox.core.TableFile;
+import io.whitefox.core.TableFileIdHashFunction;
+import io.whitefox.core.TableFileToBeSigned;
+import java.net.URI;
+import java.time.Duration;
+import java.util.Optional;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.presigner.S3Presigner;
+import software.amazon.awssdk.services.s3.presigner.model.GetObjectPresignRequest;
+import software.amazon.awssdk.services.s3.presigner.model.PresignedGetObjectRequest;
+
+public class S3FileSigner implements FileSigner {
+
+  private final S3Presigner presigner;
+  private final TableFileIdHashFunction tableFileIdHashFunction;
+
+  public S3FileSigner(S3Presigner s3Presigner, TableFileIdHashFunction tableFileIdHashFunction) {
+    this.presigner = s3Presigner;
+    this.tableFileIdHashFunction = tableFileIdHashFunction;
+  }
+
+  @Override
+  public TableFile sign(TableFileToBeSigned s) {
+    URI absPath = URI.create(s.url());
+    String bucketName = absPath.getHost();
+    String keyName = stripPrefix(absPath.getPath(), "/");
+    PresignedGetObjectRequest presignedGetObjectRequest =
+        presigner.presignGetObject(getObjectPresignRequest(bucketName, keyName));
+    return new TableFile(
+        presignedGetObjectRequest.url().toString(),
+        tableFileIdHashFunction.hash(presignedGetObjectRequest.url().toString()),
```

Comment:
this should be the hash of the original URL not the pre-signed one (that will change every time)